### PR TITLE
fix totals issue

### PR DESCRIPTION
**Pull request by jdinh8124**
## Purpose
Fix for totals from Daniel. Fixes issue where recalculation doesn't calculate when all sections are closed in OMS and fixes issue where recalculation occurs on save.

### CODE REVIEW

**Review by daniel-belcher — APPROVED** [review]

File: services/ui-src/src/measures/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 1]
```diff
@@ -1,137 +1,119 @@
...
-interface OPMProps {
-  /** Rate fields and descs from OPM */
-  OPM: Types.OtherRatesFields[];
-  /** should rate be user editable? */
-  rateReadOnly: boolean;
-  /** name for react-hook-form registration */
-  name: string;
-  rateMultiplicationValue?: number;
-  customMask?: RegExp;
-  allowNumeratorGreaterThanDenominator?: boolean;
```
Comment (jdinh8124) [inline services/ui-src/src/measures/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:39]: Code cleanup

File: services/ui-src/src/measures/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 2]
```diff
@@ -1,137 +1,119 @@
...
-  addSecondaryRegisterTag: boolean;
-  rateReadOnly: boolean;
-  performanceMeasureArray: Types.RateFields[][];
-  categories: string[];
-  majorIndex: number;
-  value: string;
-  name: string;
-  rateMultiplicationValue?: number;
-  customMask?: RegExp;
-  allowNumeratorGreaterThanDenominator: boolean;
```
Comment (jdinh8124) [inline services/ui-src/src/measures/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:66]: Additional code cleanup

File: services/ui-src/src/measures/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 3]
```diff
@@ -1,137 +1,119 @@
...
+      ) {
+        field.onChange([
+          {
+            numerator: `${tempRate.numerator ?? ""}`,
+            denominator: `${tempRate.denominator ?? ""}`,
+            rate: (!isNaN(parseFloat(tempRate.rate)) && tempRate.rate) || "",
+          },
+        ]);
+      }
+      setPrevRunWasLoad(currentRunIsLoadState);
```
Comment (jdinh8124) [inline services/ui-src/src/measures/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:116]: Added conditionals to only recalculate the total rate only if the previous useEffect run was not the initial load or there is no saved total rate value (from the API), and the current run is not the initial load state.

File: services/ui-src/src/measures/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx [hunk 4]
```diff
@@ -1,137 +1,119 @@
...
     usePerformanceMeasureContext();
   const { watch, control } = useFormContext();
+  const cleanedCategory = cleanString(category);
 
   const watchOMS = watch(omsName);
   const { field } = useController({ name: totalName, control });
-  const [prevCalcRate, setPrevCalcRate] = useState({
-    numerator: 0,
-    denominator: 0,
+  const [prevRunWasLoad, setPrevRunWasLoad] = useState(true);
```
Comment (jdinh8124) [inline services/ui-src/src/measures/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:48]: state to check that the previous run was the initial load run in the use effect

File: services/ui-src/src/measures/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx (hunk 3 above)
Comment (benjaminpaige) [reply services/ui-src/src/measures/CommonQuestions/OptionalMeasureStrat/ndrSets.tsx:116]: interesting solution. if it works im 👍  with it : )

**Review by benjaminpaige — APPROVED** [review]

**Pull request merged**